pallets/capacity/src/lib.rs

```diff
@@ -1062,6 +1068,80 @@ impl<T: Config> Pallet<T> {
 		ProviderBoostHistories::<T>::set(account, Some(boost_history));
 		Ok(())
 	}
+
+	pub(crate) fn has_unclaimed_rewards(account: &T::AccountId) -> bool {
+		let current_era = Self::get_current_era().era_index;
+		match Self::get_staking_history_for(account) {
+			Some(provider_boost_history) => {
+				match provider_boost_history.count() {
+					0usize => false,
+					// they staked before the current era, so they have unclaimed rewards.
+					1usize => provider_boost_history.get_entry_for_era(&current_era).is_none(),
+					_ => true,
```

Comment:
If the `provider_boost_history` only has values that are not claimable due to age, this would still return true I believe.

Reply:
So long as they have a non-zero history entry, that is correct and that is what you want. They will only be able to claim rewards for the last N eras where N is the maximum history retention.  I added a test that illustrates this, and subsequently found a bug where if you boost, and then wait a number of eras past the history retention, the first element of unclaimed rewards is always zeroed out for eligible balance and reward, which is wrong.

Because of this question, I also realized that if e.g. they unstaked everything with only a single entry for the current era, then the provider_boost_history should be deleted.  I added functionality for this and testing also.

There are other things that will need to be done to the history when an account claims rewards, but that will come in when the actual `claim_rewards` extrinsic is implemented.